vega_sim/scenario/fuzzed_markets/agents.py

```diff
@@ -1613,3 +1788,83 @@ def finalise(self):
             f" {self.accepted_proposals}/{self.proposals} valid governance transfer"
             " proposals."
         )
+
+
+class FuzzyRandomTransactionAgent(StateAgentWithWallet):
+    NAME_BASE = "fuzzy_random_transaction_agent"
+
+    def __init__(
+        self,
+        key_name: str,
+        wallet_name: Optional[str] = None,
+        asset_name: Optional[str] = None,
+        initial_mint: float = 1e9,
+        random_state: Optional[RandomState] = None,
+        tag: Optional[str] = None,
+    ):
+        super().__init__(wallet_name=wallet_name, key_name=key_name, tag=tag)
+
+        self.asset_name = asset_name
+        self.random_state = random_state if random_state is not None else RandomState()
+
+        self.fuzzers = {
+            proto_name: JSONHandlingProtobufGenerator(protobuf_cls.DESCRIPTOR).permute()
+            for (protobuf_cls, proto_name, _) in COMMAND_AND_TYPES
+        }
+        self.initial_mint = initial_mint
+
+    def initialise(
+        self,
+        vega: VegaService,
+        create_key: bool = True,
+        mint_key: bool = True,
+    ):
+        # Initialise wallet
+        super().initialise(vega=vega, create_key=create_key)
+
+        if self.asset_name:
+            # Get asset id
+            self.asset_id = self.vega.find_asset_id(symbol=self.asset_name)
+            if mint_key:
+                # Top up asset
+                self.vega.mint(
+                    key_name=self.key_name,
+                    asset=self.asset_id,
+                    amount=self.initial_mint,
+                    wallet_name=self.wallet_name,
+                )
+        self.vega.mint(
+            wallet_name=self.wallet_name,
+            asset=self.vega.find_asset_id(symbol="VOTE", enabled=True),
+            amount=1e4,
+            key_name=self.key_name,
+        )
+
+        self.vega.stake(
+            amount=1,
+            key_name=self.key_name,
+            wallet_name=self.wallet_name,
+        )
+
+    def step(self, vega_state):
+        weight_sum = sum(i[2] for i in COMMAND_AND_TYPES)
+        tx_name = self.random_state.choice(
+            [a[1] for a in COMMAND_AND_TYPES],
+            p=[i[2] / weight_sum for i in COMMAND_AND_TYPES],
+        )
+
+        skip_through = self.random_state.randint(1, 200)
+        fuzzer = self.fuzzers[tx_name]
+
+        # Many more protos than we can realistically fuzz, so sample through
+        # with random steps to get different ones each run
+        for _ in range(0, skip_through):
+            next(fuzzer)
+
+        fuzz_proto = next(fuzzer)
+        self.vega.submit_transaction(
+            key_name=self.key_name,
+            transaction=fuzz_proto,
+            transaction_type=tx_name,
+            wallet_name=self.wallet_name,
+        )
```

Comment:
Could we catch any request exceptions raised by this in a similar fashion to the other agents? Sometimes I'll run the fuzzing with the full wallet to debug transactions rejected by the wallet.

```
  File "/Users/caed/Projects/vega-market-sim/vega_sim/scenario/fuzzed_markets/agents.py", line 1865, in step
    self.vega.submit_transaction(
  File "/Users/caed/Projects/vega-market-sim/vega_sim/service.py", line 4028, in submit_transaction
    self.wallet.submit_transaction(
  File "/Users/caed/Projects/vega-market-sim/vega_sim/wallet/vega_wallet.py", line 322, in submit_transaction
    raise exception
  File "/Users/caed/Projects/vega-market-sim/vega_sim/wallet/vega_wallet.py", line 312, in submit_transaction
    response.raise_for_status()
  File "/Users/caed/Projects/vega-market-sim/.venv/lib/python3.9/site-packages/requests/models.py", line 1024, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://localhost:58617/api/v2/requests
```